### PR TITLE
43 minimum viable product alter database handling to sql alchemy

### PR DESCRIPTION
**Pull request by RadiationOcelot**
Removed raw SQL queries, created models using sqlalchemy, initialized database, now named `app.db` located in app folder.

Things to note:
- You can now access a User, Challenge, or Submission via `Model.query.get(id)` (e.g. `u = User.query.get(0)` will return a User object with id 0 where properties such as the username can be accessed via `u.username`)
- There is a column for password hashes in the User table
- There are functions to check if a certain User, post or challenge exists in the database.
- All times stored in the database are in Epoch
- When altering a row, retrieve the model object, change the property, then `db.session.commit()` to update the database

Things to do in the future:
- login_user() line in login() has been commented out
- We need to do one migration of the database
- (For whatever reason) sqlalchemy doesn't enforce foreign keys (or at least I couldn't figure out how to), not necessary for MVP but something to look into in the future

### CODE REVIEW

File: app/routes.py [hunk 1]
```diff
@@ -9,11 +9,12 @@
 @app.route('/index', methods=['GET'])
 def index():
     #Challenges
-    challenges = []
-    for chal in getAllChallenges():
-        challenges.append(chal.toDict())
+    challengeList = []
+    challenges = Challenge.query.all()
+    for challenge in challenges:
+        challengeList.append(challenge.toDict())
```
Comment (RedBlueCarrots) [inline app/routes.py:15]: To my understanding, everything here looks excellent.

One (very) small thing is that this refactor has eliminated the secondsToTime utility function and instead has timeLeft as an Epoch integer.

Are you able to add something here that, when looping through these challenges, uses that utility instead.

LGTM other than that though

**Review by RedBlueCarrots — CHANGES_REQUESTED** [review]
Just one minor change to replicate functionality from before

File: app/routes.py (hunk 1 above)
Comment (RadiationOcelot) [reply app/routes.py:15]: Done :)

It no longer shows an epoch integer, but the amount of time remaining that's actually readable by a human.

image

(May have to look into the order of which challenges are displayed...)

Comment (RedBlueCarrots) [reply app/routes.py:15]: Order we can worry about later in MMP, LGTM! Happy to approve

**Review by RedBlueCarrots — APPROVED** [review]

**Pull request merged**